compiler/lib/eval.ml

```diff
@@ -33,21 +33,21 @@ module Int = Int32
 
 let int_binop l f =
   match l with
-  | [ Int i; Int j ] -> Some (Int (f i j))
+  | [ Int (_, i); Int (_, j) ] -> Some (Int (Regular, f i j))
```

Comment:
Why don't we check equality of kind and forward it ?

Reply:
With JSOO, integer conversion functions are turned into the identity. So, a binary operation can have any kind of integer as argument. But the output is definitively a regular integer.